experimental/dds/tree2/src/test/feature-libraries/editable-tree-2/unboxed.spec.ts

```diff
@@ -196,17 +204,202 @@ describe("unboxed unit tests", () => {
+
+				const unboxed = unboxedTree(context, mapSchema, cursor);
+				assert.equal(unboxed.size, 0);
+			});
+
+			it("Single type", () => {
+				const builder = new SchemaBuilder("test");
+				const stringLeafSchema = builder.leaf("string", ValueSchema.String);
+				const mapSchema = builder.map("map", SchemaBuilder.fieldOptional(stringLeafSchema));
+				const rootSchema = SchemaBuilder.fieldOptional(mapSchema);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const { context, cursor } = initializeTreeWithContent(schema, {
+					foo: "Hello",
+					bar: "world",
+				});
+				cursor.firstNode(); // Root node field has 1 node; move into it
+
+				const unboxed = unboxedTree(context, mapSchema, cursor);
+				assert.equal(unboxed.size, 2);
+				assert.equal(unboxed.get(brand("foo")), "Hello");
+				assert.equal(unboxed.get(brand("bar")), "world");
+			});
+
+			it("Any type", () => {
+				const builder = new SchemaBuilder("test");
+				builder.leaf("string", ValueSchema.String);
+				const mapSchema = builder.map("map", SchemaBuilder.fieldOptional(Any));
+				const rootSchema = SchemaBuilder.fieldOptional(mapSchema);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const { context, cursor } = initializeTreeWithContent(schema, {
+					foo: "Hello",
+					bar: "world",
+				});
+				cursor.firstNode(); // Root node field has 1 node; move into it
+
+				const unboxed = unboxedTree(context, mapSchema, cursor);
+				assert.equal(unboxed.size, 2);
+
+				const foo = unboxed.get(brand("foo"));
+				assert(foo !== undefined);
+				assert.equal(foo.type, "string");
+				assert.equal(foo.value, "Hello");
+
+				const bar = unboxed.get(brand("bar"));
+				assert(bar !== undefined);
+				assert.equal(bar.type, "string");
+				assert.equal(bar.value, "world");
+			});
+
+			it("Union type", () => {
+				const builder = new SchemaBuilder("test");
+				const stringLeafSchema = builder.leaf("string", ValueSchema.String);
+				const booleanLeafSchema = builder.leaf("boolean", ValueSchema.Boolean);
+				const mapSchema = builder.map(
+					"map",
+					SchemaBuilder.fieldOptional(stringLeafSchema, booleanLeafSchema),
+				);
+				const rootSchema = SchemaBuilder.fieldOptional(mapSchema);
+				const schema = builder.intoDocumentSchema(rootSchema);
+
+				const { context, cursor } = initializeTreeWithContent(schema, {
+					foo: "Hello world",
+					bar: true,
+				});
+				cursor.firstNode(); // Root node field has 1 node; move into it
+
+				const unboxed = unboxedTree(context, mapSchema, cursor);
+				assert.equal(unboxed.size, 2);
+
+				const foo = unboxed.get(brand("foo"));
+				assert(foo !== undefined);
+				assert.equal(foo.type, "string");
+				assert.equal(foo.value, "Hello world");
+
+				const bar = unboxed.get(brand("bar"));
+				assert(bar !== undefined);
+				assert.equal(bar.type, "boolean");
+				assert.equal(bar.value, true);
+			});
+
+			// it("Recursive", () => {
```

Comment:
@CraigMacomber `mapRecursive` fails to compile. Hoping to look at this with you at some point.